### PR TITLE
🌱 Add release-1.4 related changes

### PR DESCRIPTION
**Pull request by kashifest**
Updates Makefile and README with release-1.4 branch and v1.4.0 mentions.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -23,6 +23,8 @@ cluster on top of bare metal infrastructure using Metal3.
 | v1beta1       | v1beta1             | v1.1.X        | v1.1.X       |
 | v1beta1       | v1beta1             | v1.2.X        | v1.2.X       |
 | v1beta1       | v1beta1             | v1.3.X        | v1.3.X       |
+| v1beta1       | v1beta1             | v1.4.X        | v1.4.X       |
+
```
Comment (mboukhalfa) [inline README.md:27]: extra line

**Review by tuminoid — APPROVED** [review]
/lgtm

**Pull request merged**